akka-cluster/src/main/scala/akka/cluster/Member.scala

```diff
@@ -243,18 +245,39 @@ object MemberStatus {
       Removed → Set.empty[MemberStatus])
 }
 
+object UniqueAddress extends AbstractFunction2[Address, Int, UniqueAddress] {
+
+  // for binary compatibility
+  @deprecated("Use Long UID apply instead")
+  def apply(address: Address, uid: Int) = new UniqueAddress(address, uid.toLong)
+
+}
+
 /**
  * Member identifier consisting of address and random `uid`.
  * The `uid` is needed to be able to distinguish different
  * incarnations of a member with same hostname and port.
  */
 @SerialVersionUID(1L)
-final case class UniqueAddress(address: Address, uid: Int) extends Ordered[UniqueAddress] {
-  override def hashCode = uid
+final case class UniqueAddress(address: Address, longUid: Long) extends Ordered[UniqueAddress] {
```

Comment:
I don't get it, there is such a constructor and apply method added manually in the PR.